scripts/babbage/script-context-equivalance-test.sh

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env bash
+
+# Unofficial bash strict mode.
+# See: http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -e
+set -o pipefail
+
+export WORK="${WORK:-example/work}"
+export BASE="${BASE:-.}"
+export CARDANO_CLI="${CARDANO_CLI:-cardano-cli}"
+export CARDANO_NODE_SOCKET_PATH="${CARDANO_NODE_SOCKET_PATH:-example/main.sock}"
+export TESTNET_MAGIC="${TESTNET_MAGIC:-42}"
+export UTXO_VKEY="${UTXO_VKEY:-example/utxo-keys/utxo1.vkey}"
+export UTXO_SKEY="${UTXO_SKEY:-example/utxo-keys/utxo1.skey}"
+export RESULT_FILE="${RESULT_FILE:-$WORK/result.out}"
+
+mkdir -p "$WORK"
+
+echo "Socket path: $CARDANO_NODE_SOCKET_PATH"
+echo "Socket path: $(pwd)"
+
+ls -al "$CARDANO_NODE_SOCKET_PATH"
+
+# NB: This plutus script uses a "typed" redeemer and "typed" datum.
+plutusscriptinuse="$BASE/scripts/plutus/scripts/v2/context-equivalence-test.plutus"
+# This datum hash is the hash of the typed 42
+scriptdatumhash="fcaa61fb85676101d9e3398a484674e71c45c3fd41b492682f3b0054f4cf3273"
+datumfilepath="$BASE/scripts/plutus/data/typed-42.datum"
+redeemerfilepath="$BASE/scripts/plutus/data/script-context.redeemer"
+certifyingscript="scripts/plutus/scripts/v2/stake-script.plutus"
+
+# Create certificate
+cardano-cli stake-address registration-certificate \
+  --stake-script-file "$certifyingscript" \
+  --out-file "$WORK/script.regcert"
+
+# Step 1: Create a tx output with a datum hash at the script address. In order for a tx output to be locked
+# by a plutus script, it must have a datahash. We also need collateral tx inputs so we split the utxo
+# in order to accommodate this.
+
+
+plutusscriptaddr=$($CARDANO_CLI address build --payment-script-file "$plutusscriptinuse"  --testnet-magic "$TESTNET_MAGIC")
+readonlyaddress=addr_test1vz3t3f2kgy2re66tnhgxc4t8jgylw2cqfnxdwlrq9agfmtstxxkm5
```

Comment:
What does this address correspond to?

Reply:
The input at this address will be used as a reference input. I've added a comment